### PR TITLE
Tentative windows support through clang

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@conda-forge-admin, please rerender

**Comment by serge-sans-paille** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,19 +9,20 @@ source:
...
-  skip: true  # [win and py27]
   entry_points:
     - pythran = pythran.run:run
     - pythran-config = pythran.config:run
 
 requirements:
   build:
-    - {{ compiler('cxx') }}
+    - {{ compiler('cxx') }}  # [not win]
+    - clang  # [win]
```
Comment (saraedum) [inline recipe/meta.yaml]: That's strange. Why are you not happy with the default compiler?

Comment (serge-sans-paille) [reply recipe/meta.yaml]: Visual compiler fails to compile many of pythran-generated compiler, I'm trying to use clang-cl instead (but currently fails :-/)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,19 +9,20 @@ source:
   sha256: 3744eca3e2284aee3137c8995025bde7ea76d5b78fad6d26e709a8c167c5a61d
 
 build:
-  number: 0
+  number: 1
   script:
     - "{{ PYTHON }} -c 'import os, shutil; shutil.rmtree(os.path.join(\"pythran\", \"xsimd\"))'"
     - "{{ PYTHON }} -c 'with open(\"setup.cfg\", \"w\") as s: s.write(\"[build_py]\\nno_xsimd=True\")'"
+    - "{{ PYTHON }} -c 's = open(\"pythran\\pythran-win32.cfg\", \"r\"); c = s.read(); s.close(); s = open(\"pythran\\pythran-win32.cfg\", \"w\"); s.write(c.replace(\"CC=\", \"CC=clang\").replace(\"CXX=\",\"CXX=clang++\")); s.close()'"  # [win]
```
[the remark this answers is not in the document]
Comment (serge-sans-paille) [reply recipe/meta.yaml]: +1

**Comment by saraedum** [comment]
@conda-forge-admin, please rerender

**Comment by saraedum** [comment]
@serge-sans-paille so the only thing that fails is Python 2.7 builds on Windows. Maybe we just want to disable that?

**Comment by saraedum** [comment]
Let me try this one thing that I've seen elsewhere first though…

**Comment by saraedum** [comment]
@serge-sans-paille this now skips Python 2.7 on Windows. Do you think that's acceptable?

**Comment by serge-sans-paille** [comment]
@saraedum it is! According to the log, this is still using cl.exe and not clang-cl.exe, so the patch is still not enough, I need to dig into this some more before we merge. thanks a bunch for the help!

**Comment by serge-sans-paille** [comment]
@saraedum we should rebase that on master as 0.9.4 was just released

**Comment by serge-sans-paille** [comment]
0.9.4 supports clang-cl :-)

**Comment by serge-sans-paille** [comment]
Now that pythran officially supports clang-cl, try to use it from conda

**Comment by serge-sans-paille** [comment]
@saraedum I'm not 100% happy with that version, but at least it uses clang-cl as expected.

**Pull request merged**